### ISSUE TITLE
[Feature Request] Use movie poster in notification instead of Radarr logo

### ISSUE THREAD

**Issue by KruyKaze** [Radarr/Radarr#1293]
I am referring to push notifications through Join ( the service I use) and Pushbullet etc... Is that possible? 
Thank you.

**Comment by JamzTheMan** [comment]
And or a link to IMDB would be cool. When pushed to slack, you get a nice preview... (slack pulls the info on any URL posts, so only the URL is actually sent by CP)

This is how couchpotato pushes to slack look: 

![image](https://cloud.githubusercontent.com/assets/8051654/24690095/574ed524-1990-11e7-976c-b3b4029141a7.png)

**Comment by galli-leo** [comment]
@KruyKaze Pretty sure that's impossible since we can only change the icon at the time of creating our "bot account".

@JamzTheMan Can you open a new request for that one?

**Comment by KruyKaze** [comment]
@galli-leo actually Join lets you define an icon for each notification. 
Thanks

**Comment by KruyKaze** [comment]
I looked it up and it is just a matter of adding "?icon=www.XYZ.jpg&" to the notification 
Thanks

**Comment by kjagiello** [comment]
Hello! I've started to implement enhanced Slack notifications based on this issue and #1532. I could have settled on an easy way of implementing it and simply sending a message containing an IMDB link and let Slack show summary of the page, but I wanted to try to utilise the Slack message formatting engine and came up with this:

<img width="663" alt="screen shot 2017-12-18 at 20 57 40" src="https://user-images.githubusercontent.com/74944/34125300-1a89e25c-e436-11e7-9d02-c75c3e31d3cc.png">

The problem is however the poster image. I cannot really use the one in Radarr as the Radarr may be on a network that is not accessible by Slack. I don't want to upload the poster image from Radarr either, as I'm using the free version and the upload space is limited. So what I need is to be able to easily generate an reliable, public URL to the poster image. AFAIK there is no easy way of getting the poster image from neither IMDB or TVDB. So I wanted to ask you that know the system internals inside out and know if there is an easy of achieving this?

**Comment by thomaswelton** [comment]
@kjagiello Have you seen this? http://www.omdbapi.com
Using their API you can get JSON data for a movie using an IMDB id and it'll return a bunch of info including the poster image.

But also, on one of the messages I posted to slack it automatically fetched that URL and added the movie poster in slack, but for 99% it does not do that. I'm wondering if adding `"unfurl_links": true` will make slack fetch all the URL previews

```
We will automatically fetch and create attachments for certain well-known media URLs, but we will not unfurl links to primarily text-based content, unless you specify in your payload that you want that. If you do, add "unfurl_links": true to your main JSON payload. Example links that will unfurl without the flag:
```

**Comment by bonswouar** [comment]
This would be a nice to have, although using imdb image url would mean one call on amazon webservers for each receiving device, which is not really wanted imo.

I personally use ntfy, and we can easily attach an image to a notification: https://docs.ntfy.sh/publish/#attach-local-file
I don't know about other notification systems but I think this should be the recommended way to do it

**Comment by CoolnsX** [comment]
@bonswouar Exactly what I wanted too..

Is there any PR for this? If not, I want to do it, even though idk .net but I'll try

**Comment by bonswouar** [comment]
@CoolnsX There's no PR afaik, feel free to propose one, that should be quite easy to implement for ntfy (I don't know about other services though)

**Comment by CoolnsX** [comment]
Yeah that's why I am willing to take it. I will do it this weekend though.

**Comment by vampywiz17** [comment]
Hello there, these feature working on telegram, but not on ntfy. it will be released?

**Comment by omarahmed786** [comment]
> [@CoolnsX](https://github.com/CoolnsX) There's no PR afaik, feel free to propose one, that should be quite easy to implement for ntfy (I don't know about other services though)

Hello friend. Would you mind walking me through how to add this to Ntfy? I'm currently using a custom script with Ntfy that allows me to deeplink into the nzb360 app on my Android phone (the primary way I engage with Ntfy notifications of my server).

Right now when I click on the notification on my android phone:
<img width="1338" height="826" alt="Image" src="https://github.com/user-attachments/assets/2f7d4629-4783-4d34-918c-c071065ed9da" />
I am taken to this view:
<img width="1344" height="2992" alt="Image" src="https://github.com/user-attachments/assets/58f89fa5-5636-4247-a4b2-c711b06661de" />

using this script:
```
#!/bin/bash

NTFY_URL="http://X.X.X.X:PORT"
NTFY_TOPIC="radarr"
NTFY_TOKEN="TOKEN"

EVENT_TYPE="$radarr_eventtype"

# Friendly event name mapping
declare -A FRIENDLY_EVENT_NAMES=(
  [Download]="Movie Imported"
  [Grab]="Movie Grabbed"
  [Rename]="File Renamed"
  [MovieDelete]="Movie Deleted"
  [MovieAdded]="Movie Added"
  [MovieFileDelete]="Movie File Deleted"
  [ApplicationUpdate]="Application Updated"
  [Test]="Test Notification"
)

FRIENDLY_NAME="${FRIENDLY_EVENT_NAMES[$EVENT_TYPE]:-$EVENT_TYPE}"

if [[ "$EVENT_TYPE" == "Test" ]]; then
    TITLE="Radarr - ${FRIENDLY_NAME}"
    BODY="This is a test message from Radarr"
    CLICK_URL="nzb360://radarr"

elif [[ "$EVENT_TYPE" == "ApplicationUpdate" ]]; then
    TITLE="Radarr - ${FRIENDLY_NAME}"
    BODY="Radarr updated from ${radarr_update_oldversion} to ${radarr_update_newversion}"
    CLICK_URL="http://radarr.domain.com/settings/general"

else
    MOVIE_TITLE="$radarr_movie_title"
    MOVIE_YEAR="$radarr_movie_year"
    MOVIE_ID="$radarr_movie_id"
    QUALITY="$radarr_moviefile_quality"
    
    TITLE="Radarr - ${FRIENDLY_NAME}"
    BODY="${MOVIE_TITLE} (${MOVIE_YEAR})"
    CLICK_URL="nzb360://radarr?movieId=${MOVIE_ID}"
fi

curl -X POST "${NTFY_URL}/${NTFY_TOPIC}" \
      -H "Authorization: Bearer ${NTFY_TOKEN}" \
      -H "Icon: https://cdn.jsdelivr.net/gh/selfhst/icons/png/radarr.png" \
      -H "Title: ${TITLE}" \
      -H "Click: ${CLICK_URL}" \
      -d "${BODY}"
```

And it works amazingly. But as you can see in the first screenshot, I would really love for an image to be attached to it as well of the movie (or tv show for sonarr). Do you know where I can find the relevant poster image file so that I can add it to my curl command via
`-H "Filename: flower.jpg" \`

**Comment by bonswouar** [comment]
@omarahmed786 as far as I know the MediaCover isn't part of custom script exposed params, although you could use Radarr API directly.

I haven't tested it with your example script, but it should look something like:
```bash
# ...
## Download poster img to a temporary file
TMP_POSTER_FILE="cover_${MOVIE_ID}.jpg"
RADARR_URL="https://your.radarr.url"
RADARR_API_KEY="yourRadarrApiKey"
curl "${RADARR_URL}/api/v3/mediacover/${MOVIE_ID}/poster.jpg?apikey=${RADARR_API_KEY}" --output "${TMP_POSTER_FILE}"

curl -X POST "${NTFY_URL}/${NTFY_TOPIC}" \
      -H "Authorization: Bearer ${NTFY_TOKEN}" \
      -H "Icon: https://cdn.jsdelivr.net/gh/selfhst/icons/png/radarr.png" \
      -H "Title: ${TITLE}" \
      -H "Click: ${CLICK_URL}" \
      -d "${BODY}" \
      -H "Filename: poster.jpg" \
      -H "${TMP_POSTER_FILE}"
rm "${TMP_POSTER_FILE}"
```
